### PR TITLE
#569: move homepage queries to page level

### LINKED ISSUE #569: pull all homepage queries into a single query at the top level (including marketplaces)

### PR DESCRIPTION
**Pull request by austinmilt**
This PR moves all homepage queries to a custom hook that is executed at the page level and passes props down to sections.

## Start in `home.hooks.ts`

# Changes
- use graph QL fragments to organize request data (`feed.graphql`, `home.graphql`)
- move all graph QL homepage requests into a single request (`home.graphql`, `featuredBuyNowListings.graphql`, `featuredProfiles.graphql`, `marketplacePreview.graphql`, `profilePreview.graphql`)
- created custom data interfaces for homepage sections to define their data needs (`ProfilePreviewCard.tsx`, `FeaturedAuctionsSection.tsx`, `FeaturedBuyNowListingsSection.tsx`, `FeaturedProfilesSection.tsx`, `FeaturedMarketplacesSection.tsx`)
- created a custom hook to wrap requests and transform into the correct format for components (`home.hooks.ts`, `index.tsx`, `nft.tsx`) which required making a generic refetch interface for several components (`SellForm.tsx`, `UpdateSellForm.tsx`, `UpdateOfferForm.tsx`
- use custom hook in homepage (`index.tsx`) and removed queries from sections (`FeaturedAuctionsSection.tsx`, `FeaturedBuyNowListingsSection.tsx`, `FeaturedProfilesSection.tsx`, `FeaturedMarketplacesSection.tsx`)

closes #569

### CODE REVIEW

File: src/views/home/home.hooks.ts [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+  refetch: () => void;
+}
+
+//TODO remove once other profiles have enough followers to preclude this one in the backend
+const DISALLOWED_PROFILES: string[] = ['ho1aVYd4TDWCi1pMqFvboPPc3J13e4LgWkWzGJpPJty'];
+
+export function useHomeQueryWithTransforms(
+  userWallet: PublicKey | null,
+  featuredProfileLimit: number,
+  featuredBowNowLimit: number,
```
Comment (kristianeboe) [inline src/views/home/home.hooks.ts:32]: featuredBuyNowLimit?

File: src/views/home/home.hooks.ts [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+  const loading: boolean = queryContext.loading || featuredAuctionContext.loading;
+
+  const data: HomeData | undefined = useMemo(() => {
+    let result: HomeData | undefined;
+    if (!loading && queryContext.called && !queryContext.error) {
+      result = {
+        feedEvents: transformHeroSectionData(queryContext.data?.feedEvents),
+        featuredProfiles: transformFeaturedProfiles(queryContext.data?.followWallets),
+        featuredMarketplaces: transformFeaturedMarketplaces(
+          queryContext.data?.featuredMarketplace0,
```
Comment (kristianeboe) [inline src/views/home/home.hooks.ts:56]: I'd prefer this as a list, but no biggie

File: src/views/home/home.hooks.ts [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+  }, [nListings, applyListingFilterAndSort]);
+
+  return {
+    loading: loading,
+    data: result,
+    refetch: () => {}
+  }
+}
+
+function notNullish(value: any, name: string): void {
```
Comment (kristianeboe) [inline src/views/home/home.hooks.ts:285]: These are some nice utils! Should probably be in their own file somewhere

**Review by milorue — APPROVED** [review]
Besides @kristianeboe's comments this looks good to me

**Review by kristianeboe — APPROVED** [review]
I think this is a big improvement over the current setup of the home page. Let's merge!

File: src/views/home/home.hooks.ts (hunk 2 above)
Comment (austinmilt) [reply src/views/home/home.hooks.ts:56]: oh, whoops, I forgot I added a query for multiple marketplaces :D

**Pull request merged**